### ISSUE TITLE
Valet install uncaught error

### ISSUE THREAD

**Issue by namnguyenvn** [cpriego/valet-linux#38]
When I ran "valet install" I got the following error:
`namnguyen@namnguyen-linux:~/.composer$ valet install
PHP Fatal error:  Uncaught ReflectionException: Class 0 does not exist in /home/namnguyen/.composer/vendor/illuminate/container/Container.php:734
Stack trace:
#0 /home/namnguyen/.composer/vendor/illuminate/container/Container.php(734): ReflectionClass->__construct('0')
#1 /home/namnguyen/.composer/vendor/illuminate/container/Container.php(629): Illuminate\Container\Container->build(0, Array)
#2 /home/namnguyen/.composer/vendor/cpriego/valet-linux/cli/includes/helpers.php(58): Illuminate\Container\Container->make(0)
#3 /home/namnguyen/.composer/vendor/cpriego/valet-linux/cli/Valet/Valet.php(141): resolve(0)
#4 [internal function]: Valet\Valet->Valet\{closure}(0, 'Valet\\PackageMa...')
#5 /home/namnguyen/.composer/vendor/illuminate/support/Arr.php(157): call_user_func(Object(Closure), 0, 'Valet\\PackageMa...')
#6 /home/namnguyen/.composer/vendor/illuminate/support/Collection.php(282): Illuminate\Support\Arr::first(Array, Object(Closure), Object(Closure))
#7 /home/namnguyen/.composer/vendor/cpriego/valet-linux/cli/Valet/Valet.php in /home/namnguyen/.composer/vendor/illuminate/container/Container.php on line 734
`

My environment details are:
Ubuntu 16.10
PHP 7.0.15
composer 1.4.1

Please help me figure out what happened. Thank you so much!

**Comment by cpriego** [comment]
Could you please execute this command?
```
test -d ~/.composer && bash ~/.composer/vendor/bin/valet install || bash ~/.config/composer/vendor/bin/valet install
```

**Comment by namnguyenvn** [comment]
Hi Carlos, 

I still have the same error:

`namnguyen@namnguyen-linux:~$ test -d ~/.composer && bash ~/.composer/vendor/bin/valet install || bash ~/.config/composer/vendor/bin/valet install
[sudo] password for namnguyen: 
PHP Fatal error:  Uncaught ReflectionException: Class 0 does not exist in /home/namnguyen/.composer/vendor/illuminate/container/Container.php:734
Stack trace:
#0 /home/namnguyen/.composer/vendor/illuminate/container/Container.php(734): ReflectionClass->__construct('0')
#1 /home/namnguyen/.composer/vendor/illuminate/container/Container.php(629): Illuminate\Container\Container->build(0, Array)
#2 /home/namnguyen/.composer/vendor/cpriego/valet-linux/cli/includes/helpers.php(58): Illuminate\Container\Container->make(0)
#3 /home/namnguyen/.composer/vendor/cpriego/valet-linux/cli/Valet/Valet.php(141): resolve(0)
#4 [internal function]: Valet\Valet->Valet\{closure}(0, 'Valet\\PackageMa...')
#5 /home/namnguyen/.composer/vendor/illuminate/support/Arr.php(157): call_user_func(Object(Closure), 0, 'Valet\\PackageMa...')
#6 /home/namnguyen/.composer/vendor/illuminate/support/Collection.php(282): Illuminate\Support\Arr::first(Array, Object(Closure), Object(Closure))
#7 /home/namnguyen/.composer/vendor/cpriego/valet-linux/cli/Valet/Valet.php in /home/namnguyen/.composer/vendor/illuminate/container/Container.php on line 734
bash: /home/namnguyen/.config/composer/vendor/bin/valet: No such file or directory`

Even when I execute only .composer/vendor/bin/valet:
`namnguyen@namnguyen-linux:~/.composer/vendor/bin$ test -d ~/.composer && bash ~/.composer/vendor/bin/valet install
PHP Fatal error:  Uncaught ReflectionException: Class 0 does not exist in /home/namnguyen/.composer/vendor/illuminate/container/Container.php:734
Stack trace:
#0 /home/namnguyen/.composer/vendor/illuminate/container/Container.php(734): ReflectionClass->__construct('0')
#1 /home/namnguyen/.composer/vendor/illuminate/container/Container.php(629): Illuminate\Container\Container->build(0, Array)
#2 /home/namnguyen/.composer/vendor/cpriego/valet-linux/cli/includes/helpers.php(58): Illuminate\Container\Container->make(0)
#3 /home/namnguyen/.composer/vendor/cpriego/valet-linux/cli/Valet/Valet.php(141): resolve(0)
#4 [internal function]: Valet\Valet->Valet\{closure}(0, 'Valet\\PackageMa...')
#5 /home/namnguyen/.composer/vendor/illuminate/support/Arr.php(157): call_user_func(Object(Closure), 0, 'Valet\\PackageMa...')
#6 /home/namnguyen/.composer/vendor/illuminate/support/Collection.php(282): Illuminate\Support\Arr::first(Array, Object(Closure), Object(Closure))
#7 /home/namnguyen/.composer/vendor/cpriego/valet-linux/cli/Valet/Valet.php in /home/namnguyen/.composer/vendor/illuminate/container/Container.php on line 734
`

**Comment by cpriego** [comment]
Hi @namnguyenvn could you please make steps 2 and 3 and write back the results? I need to see if there are any conflicts with you installed packages.

**Comment by namnguyenvn** [comment]
Hi @cpriego ,
Here is my step 2 result:

namnguyen@namnguyen-linux:~$ composer global show
Changed current directory to /home/namnguyen/.composer
acacha/adminlte-laravel-installer   2.0.2   Adminlte-laravel application installer.
acacha/llum                         0.7     Llum (light in catalan language) illuminates your Laravel projects.
container-interop/container-interop 1.2.0   Promoting the interoperability of container objects (DIC, SL, etc.)
cpriego/valet-linux                 v2.0.6  A more enjoyable local development experience for Linux.
doctrine/inflector                  v1.1.0  Common String Manipulations with regard to casing and singular/plural rules.
friendsofphp/php-cs-fixer           v1.12.1 A tool to automatically fix PHP code style
guzzlehttp/guzzle                   5.3.0   Guzzle is a PHP HTTP client library and framework for building RESTful web service clients
guzzlehttp/ringphp                  1.1.0   Provides a simple API and specification that abstracts away the details of HTTP into a single PHP fu...
guzzlehttp/streams                  3.0.0   Provides a simple abstraction over streams of data
illuminate/config                   v5.2.21 The Illuminate Config package.
illuminate/container                v5.2.45 The Illuminate Container package.
illuminate/contracts                v5.2.45 The Illuminate Contracts package.
illuminate/filesystem               v5.2.21 The Illuminate Filesystem package.
illuminate/support                  v5.2.21 The Illuminate Support package.
laravel/installer                   v1.2.1  Laravel application installer.
mnapoli/silly                       1.5.1   Silly CLI micro-framework based on Symfony Console
nategood/httpful                    0.2.20  A Readable, Chainable, REST friendly, PHP HTTP Client
php-di/invoker                      1.3.3   Generic and extensible callable invoker
psr/container                       1.0.0   Common Container Interface (PHP FIG PSR-11)
psr/log                             1.0.2   Common interface for logging libraries
react/promise                       v2.2.1  A lightweight implementation of CommonJS Promises/A for PHP
sebastian/diff                      1.4.1   Diff implementation
squizlabs/php_codesniffer           2.6.0   PHP_CodeSniffer tokenizes PHP, JavaScript and CSS files and detects violations of a defined set of c...
symfony/console                     v2.8.18 Symfony Console Component
symfony/debug                       v3.0.9  Symfony Debug Component
symfony/event-dispatcher            v3.1.4  Symfony EventDispatcher Component
symfony/filesystem                  v3.1.4  Symfony Filesystem Component
symfony/finder                      v3.0.9  Symfony Finder Component
symfony/polyfill-mbstring           v1.3.0  Symfony polyfill for the Mbstring extension
symfony/process                     v2.8.18 Symfony Process Component
symfony/stopwatch                   v3.1.4  Symfony Stopwatch Component
tightenco/collect                   v5.4.14 Collect - Illuminate Collections as a separate package.


And here is my step 3 result:
namnguyen@namnguyen-linux:~$ composer global remove cpriego/valet-ubuntu
Changed current directory to /home/namnguyen/.composer
cpriego/valet-ubuntu is not required in your composer.json and has not been removed
Package "cpriego/valet-ubuntu" listed for update is not installed. Ignoring.
Loading composer repositories with package information
Updating dependencies (including require-dev)
Nothing to install or update
Generating autoload files
namnguyen@namnguyen-linux:~$ composer global remove cpriego/valet-linux
Changed current directory to /home/namnguyen/.composer
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 0 installs, 0 updates, 8 removals
  - Removing cpriego/valet-linux (v2.0.6)
  - Removing illuminate/container (v5.2.45)
  - Removing mnapoli/silly (1.5.1)
  - Removing php-di/invoker (1.3.3)
  - Removing container-interop/container-interop (1.2.0)
  - Removing psr/container (1.0.0)
  - Removing nategood/httpful (0.2.20)
  - Removing tightenco/collect (v5.4.14)
Writing lock file
Generating autoload files
namnguyen@namnguyen-linux:~$ composer global require cpriego/valet-linux
Changed current directory to /home/namnguyen/.composer
Using version ^2.0 for cpriego/valet-linux
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 8 installs, 0 updates, 0 removals
  - Installing tightenco/collect (v5.4.14): Loading from cache
  - Installing nategood/httpful (0.2.20): Loading from cache
  - Installing psr/container (1.0.0): Loading from cache
  - Installing container-interop/container-interop (1.2.0): Loading from cache
  - Installing php-di/invoker (1.3.3): Loading from cache
  - Installing mnapoli/silly (1.5.1): Loading from cache
  - Installing illuminate/container (v5.2.45): Loading from cache
  - Installing cpriego/valet-linux (v2.0.6): Loading from cache
Writing lock file
Generating autoload files

Thank you!

**Comment by cpriego** [comment]
Ok, I can see there are various packages that haven't been updated including Valet itself. Update everything with

```
composer global update
```

With a little luck that should fix your issues

**Comment by namnguyenvn** [comment]
Thank you! It works!

**Comment by arnab682** [comment]
How can I solve this...???

arnab@arnab-Inspiron-3521:~$ composer global require genesisweb/valet-linux-plus
Changed current directory to /home/arnab/.config/composer
Using version ^1.5 for genesisweb/valet-linux-plus
./composer.json has been updated
Running composer update genesisweb/valet-linux-plus
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

Problem 1
- genesisweb/valet-linux-plus[1.5, ..., 1.5.3] require illuminate/container ~5.3 -> found illuminate/container[v5.3.0, ..., v5.8.36] but the package is fixed to v8.63.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
- Root composer.json requires genesisweb/valet-linux-plus ^1.5 -> satisfiable by genesisweb/valet-linux-plus[1.5, 1.5.1, 1.5.3].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

**Comment by marcellopato** [comment]
Not working for me!

 ✘ marcellopato@pato  ~  composer global update
Changed current directory to /home/marcellopato/.config/composer
Loading composer repositories with package information
Updating dependencies
Nothing to modify in lock file
Installing dependencies from lock file (including require-dev)
Nothing to install, update or remove
Generating autoload files
19 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
No security vulnerability advisories found.
 marcellopato@pato  ~  composer global require cpriego/valet-linux
Changed current directory to /home/marcellopato/.config/composer
./composer.json has been updated
Running composer update cpriego/valet-linux
Loading composer repositories with package information
Updating dependencies
Nothing to modify in lock file
Writing lock file
Installing dependencies from lock file (including require-dev)
Nothing to install, update or remove
Generating autoload files
19 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
No security vulnerability advisories found.
Using version ^2.3 for cpriego/valet-linux
 marcellopato@pato  ~  valet install                              
zsh: command not found: valet


Linux Ubuntu 20.24
